contracts/src/token/erc721/receiver.rs

```diff
@@ -2,33 +2,22 @@
 //! ERC-721 token transfers.
 #![allow(missing_docs)]
 #![cfg_attr(coverage_nightly, coverage(off))]
-use alloc::vec;
+use alloc::{vec, vec::Vec};
 
-use stylus_sdk::prelude::sol_interface;
+use alloy_primitives::{Address, FixedBytes, U256};
+use openzeppelin_stylus_proc::interface_id;
+use stylus_sdk::{abi::Bytes, function_selector, prelude::*};
+
+/// The expected value returned from [`IErc721Receiver::on_erc721_received`].
+pub const RECEIVER_FN_SELECTOR: [u8; 4] =
+    function_selector!("onERC721Received", Address, Address, U256, Bytes,);
```

Comment:
```suggestion
pub const RECEIVER_FN_SELECTOR: FixedBytes<4> =
    FixedBytes::new(function_selector!("onERC721Received", Address, Address, U256, Bytes,));
```
Maybe it would be more efficient to change type of `RECEIVER_FN_SELECTOR`, since this is the type that will almost always be used. Users then don't need to do [manual conversions](https://github.com/OpenZeppelin/rust-contracts-stylus/pull/743/files#diff-446a0b6f6e7b54efb0731bdd5a16768d33d0d343aa5f13f1946fdcb571f6e457R10) each time and can just use it directly (in the linked code, the `const SELECTOR` becomes reduntant.